### PR TITLE
Task-7 and Task-8 - Add logic and Refactor code

### PR DESCRIPTION
**Pull request by LovchevaKate**
[Task-7: Add logic to your app](https://godelonline.sharepoint.com/Functions/LearningAndDevelopment/SitePages/Node---Proxy-Server.aspx?csf=1&web=1&e=Z6bZS6&OR=Teams-HL&CT=1728648202018&clickparams=eyJBcHBOYW1lIjoiVGVhbXMtRGVza3RvcCIsIkFwcFZlcnNpb24iOiI0OS8yNDA5MDEwMTQyMSIsIkhhc0ZlZGVyYXRlZFVzZXIiOmZhbHNlfQ%3D%3D&CID=019f519d-57cf-4606-9f98-dfd459e75b28#task-7-add-logic-to-your-app)

[Task-8: Refactor code](https://godelonline.sharepoint.com/Functions/LearningAndDevelopment/SitePages/Node---Proxy-Server.aspx?csf=1&web=1&e=Z6bZS6&OR=Teams-HL&CT=1728648202018&clickparams=eyJBcHBOYW1lIjoiVGVhbXMtRGVza3RvcCIsIkFwcFZlcnNpb24iOiI0OS8yNDA5MDEwMTQyMSIsIkhhc0ZlZGVyYXRlZFVzZXIiOmZhbHNlfQ%3D%3D&CID=019f519d-57cf-4606-9f98-dfd459e75b28#task-8-refactor-code)

Acceptance criteria:

- Add logic to the base endpoint /meteors

Acceptance criteria:

- Divide the application into layers
- add the Delivery layer
- add route to the Delivery layer
- add the UseCases layer
- add a Repository layer

### CODE REVIEW

File: config.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import dotenv from "dotenv";
+
+dotenv.config();
+
+export const config = {
+  nasaApiKey: process.env.NASA_API_KEY,
+  nasaApiUrl: process.env.NASA_API_URL,
```
Comment (MaksimShachykau) [inline config.js]: I would like to suggest using an object destructor here.
`const {NASA_API_KEY, NASA_API_URL} = process.env`

File: useCases/fetchMeteors.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import getMeteors from "../repository/meteorRepository.js";
+
+const fetchMeteors = async () => {
+  let result = await getMeteors();
```
Comment (MaksimShachykau) [inline useCases/fetchMeteors.js]: why `let`?

File: useCases/fetchMeteors.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+        2,
+      is_potentially_hazardous_asteroid: item.is_potentially_hazardous_asteroid,
+      close_approach_date_full:
+        item.close_approach_data[0].close_approach_date_full,
+      relative_velocity:
+        item.close_approach_data[0].relative_velocity.kilometers_per_second,
+    }))
+  );
+};
+
```
Comment (MaksimShachykau) [inline useCases/fetchMeteors.js]: there is too much logic here, I think you could create some helpers for this, like the getDiameter function. and you can create a function to extract the close approach data
`relative_velocity` and `close_approach_date_full`

File: repository/meteorRepository.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import axios from "axios";
+import { config } from "../config.js";
+
```
Comment (MaksimShachykau) [inline repository/meteorRepository.js:3]: you can get data from config by object destructor 
`const {nasaApiUrl, ...} = config` 
in this case you don’t have to write `config.key` each time

File: config.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+  monday: "2024-10-07",
+  friday: "2024-10-11",
```
Comment (MaksimShachykau) [inline config.js]: It's better to use constants for strings
ex. `const seventhOfOctober = '2024-10-07'`

File: helper/meteorHelper.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    (diameter.meters.estimated_diameter_min +
+      diameter.meters.estimated_diameter_max) /
+    2
+  );
+};
+
+const extractCloseApproachData = (closeApproachData) => {
+  return {
+    relative_velocity:
+      closeApproachData[0].relative_velocity.kilometers_per_second,
```
Comment (MaksimShachykau) [inline helper/meteorHelper.js:35]: what if closeApproachData is empty? (can fix in next PRs)

File: repository/meteorRepository.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import axios from "axios";
+import { NASA_API_KEY, NASA_API_URL } from "../config.js";
+
+const seventhOfOctober = "2024-10-07";
+const eleventhOfOctober = "2024-10-11";
```
Comment (MaksimShachykau) [inline repository/meteorRepository.js:5]: best practices for naming constants is using UPPERCASE f.e.` const SEVENTH_OF_OCTOBER = ""` (can fix in next PRs)

**Review by MaksimShachykau — APPROVED** [review]

File: helper/meteorHelper.js (hunk 6 above)
Comment (LovchevaKate) [reply helper/meteorHelper.js:35]: you mean add something like this:

`if (!closeApproachData || closeApproachData.length === 0) {
    return {
      relative_velocity: null,
      close_approach_date_full: null,
    };
  }`

**Pull request merged**